larpix/larpix_hpt.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import absolute_import
+
+from math import fmod
+
+def get_ns(time_in_sec):
+    return int(fmod(time_in_sec * 1e9, 1e9))
+
+def get_s(time_in_sec):
+    return int(time_in_sec)
+
+class larpix_hpt(object):
+    '''Class for precision time'''
+    larpix_clk_freq = 5e6 #Hz
+    larpix_offset_d = 2**24 #clk cycles
+    offset = {}
+    ref_time = None
+    prev_time = {}
+
+    def __init__(self, s=0, ns=0, cpu_time=0, adc_time=0):
+        '''Constructor'''
+        self.s = int(s)
+        self.ns = int(fmod(ns,1e9))
+        self.cpu_time = cpu_time
+        self.adc_time = int(adc_time)
+
+    def __str__(self):
+        string = 's: %d' % self.s
+        string += ' ns: %d' % self.ns
+        string += ' cpu_time: %f' % self.cpu_time
+        string += ' adc_time: %d' % self.adc_time
+        return string
+
+    def __repr__(self):
+        return str(self)
+
+    def __eq__(self, hpt):
+        return self.s == hpt.s and self.ns == hpt.ns
+
+    def __gt__(self, hpt):
+        if self.s > hpt.s:
+            return True
+        elif self.s == hpt.s:
+            return self.ns > hpt.ns
+        else:
+            return False
+
+    def __lt__(self, hpt):
+        return not (self > hpt) and not (self == hpt)
+
+    def __add__(self, hpt):
+        sum_ns = int(fmod(self.ns + hpt.ns, 1e9))
```

Comment:
I don't like the idea of converting to and from floats here since precision is key.

Easiest adjustment is to use the integer mod operator ``%`` and just type out ``1e9`` as an int, i.e. ``1000000000``. Same applies for the next line and for ``__sub__``